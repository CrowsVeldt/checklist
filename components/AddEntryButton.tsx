import { AntDesign } from "@expo/vector-icons";
import { Pressable, StyleSheet } from "react-native";

export default function AddEntryButton({ add }: { add: any }) {
  return (
    <Pressable
      style={({ pressed }) =>
        pressed
          ? [styles.addEntryButton, styles.buttonPressed]
          : styles.addEntryButton
      }
      onPress={add}
    >
      <AntDesign name="plus" size={30} color="black" />
    </Pressable>
  );
}

const styles = StyleSheet.create({
  addEntryButton: {
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 1,
    borderColor: "black",
    width: 40,
    height: 40,
    marginVertical: 10,
  },
  buttonPressed: {
    backgroundColor: "gray",
  },
});

// setEntries([
// ...entries,
// {
// id: randomUUID(),
// status: false,
// title: "New item",
// required: false,
// parentTo: [],
// },
// ]);
