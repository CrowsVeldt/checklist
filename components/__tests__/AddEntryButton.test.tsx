import { render } from "@testing-library/react-native";
import AddEntryButton from "../AddEntryButton";

describe("<AddEntryButton />", () => {
  test("AddEntryButton renders correctly", () => {
    const tree = render(<AddEntryButton add={undefined} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
