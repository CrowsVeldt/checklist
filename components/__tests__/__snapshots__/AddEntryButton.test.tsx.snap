// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddEntryButton /> AddEntryButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 25,
      "borderWidth": 1,
      "height": 40,
      "justifyContent": "center",
      "marginVertical": 10,
      "width": 40,
    }
  }
>
  <Text />
</View>
`;
